src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Tests/Requests/DataSetQueryCriteriaNotValidatorTests.cs

```diff
@@ -0,0 +1,283 @@
+            {
+                Filters = new DataSetQueryCriteriaFilters
+                {
+                    Eq = ""
+                },
+            },
+        };
+
+        _validator.TestValidate(query)
+            .ShouldHaveValidationErrorFor("Not.Filters.Eq");
+    }
+
+    [Fact]
+    public void Condition_Empty_Failure()
+    {
+        var query = new DataSetQueryCriteriaNot
+        {
+            Not = new DataSetQueryCriteriaOr
+            {
+                Or = [],
+            },
+        };
+
+        _validator.TestValidate(query)
+            .ShouldHaveValidationErrorFor("Not.Or");
+    }
+
+    [Fact]
+    public void Condition_SingleFacets_Success()
+    {
+        var query = new DataSetQueryCriteriaNot
+        {
+            Not = new DataSetQueryCriteriaOr
+            {
+                Or =
+                [
+                    new DataSetQueryCriteriaFacets
+                    {
+                        Filters = new DataSetQueryCriteriaFilters
+                        {
+                            Eq = "12345"
+                        },
+                    },
+                ],
+            },
+        };
+
+        _validator.TestValidate(query)
+            .ShouldNotHaveAnyValidationErrors();
+    }
+
+    [Fact]
+    public void Condition_SingleFacets_Failure()
+    {
+        var query = new DataSetQueryCriteriaNot
+        {
+            Not = new DataSetQueryCriteriaOr
+            {
+                Or =
+                [
+                    new DataSetQueryCriteriaFacets
+                    {
+                        Filters = new DataSetQueryCriteriaFilters
+                        {
+                            Eq = new string('x', 11)
+                        },
+                    },
+                ],
+            },
+        };
+
+        _validator.TestValidate(query)
+            .ShouldHaveValidationErrorFor("Not.Or[0].Filters.Eq");
+    }
+
+    [Fact]
+    public void Condition_MultipleFacets_Success()
+    {
+        var query = new DataSetQueryCriteriaNot
+        {
+            Not = new DataSetQueryCriteriaAnd
+            {
+                And =
+                [
+                    new DataSetQueryCriteriaFacets
+                    {
+                        Filters = new DataSetQueryCriteriaFilters
+                        {
+                            Eq = "12345"
+                        },
+                    },
+                    new DataSetQueryCriteriaFacets
+                    {
+                        TimePeriods = new DataSetQueryCriteriaTimePeriods
+                        {
+                            Gte = new DataSetQueryTimePeriod { Period = "2020/2021", Code = "AY" }
+                        },
+                    },
+                ],
+            },
+        };
+
+        _validator.TestValidate(query)
+            .ShouldNotHaveAnyValidationErrors();
+    }
+
+    [Fact]
+    public void Condition_MultipleSingleFacets_Failure()
```

Comment:
Maybe rename to `Condition_MultipleFacets_Failure` to be consistent with elsewhere